recipe_scrapers/_utils.py

```diff
@@ -229,15 +261,13 @@ def change_keys(obj, convert):
     Credit: StackOverflow user 'baldr'
     (https://web.archive.org/web/20201022163147/https://stackoverflow.com/questions/11700705/python-recursively-replace
         -character-in-keys-of-nested-dictionary/33668421)
+
+    Note: with modifications applied.
     """
-    if isinstance(obj, (str, int, float)):
-        return obj
     if isinstance(obj, dict):
-        new = obj.__class__()
-        for k, v in obj.items():
-            new[convert(k)] = change_keys(v, convert)
+        return {convert(k): change_keys(v, convert) for k, v in obj.items()}
     elif isinstance(obj, (list, set, tuple)):
-        new = obj.__class__(change_keys(v, convert) for v in obj)
+        type_constructor = type(obj)
+        return type_constructor(change_keys(item, convert) for item in obj)
```

Comment:
```suggestion
        cls = type(obj)
        return cls(change_keys(item, convert) for item in obj)
```

(`cls` seems to be a Pythonic way to refer to the name of a class, when provided/stored as a variable)